core-utilities/src/main/java/edu/illinois/cs/cogcomp/core/datastructures/textannotation/View.java

```diff
@@ -392,6 +392,25 @@ private View createRestrictedView(ITransformer<View, Double> scoreTransformer) {
         return list;
     }
 
+    /**
+     * Given char-begin-offset and char-end-offset, it returns the constituents with some overlap with the input char offsets.
+     * @param charStart the begin char index
+     * @param charEnd the end char index (one-past-the-end indexing)
+     */
+    public List<Constituent> getConstituentsOverlappingCharSpan(int charStart, int charEnd) {
+        assert charStart <= charEnd: "The start offset should be smaller than the end offset";
```

Comment:
When an assertion fails, an AssertionError is thrown with the message statement. So I think it is similar to throwing an Exception instance (like RuntimeException). I don't know if there are any limitations of using assertions.